archunit/src/test/java/com/tngtech/archunit/core/domain/JavaClassTest.java

```diff
@@ -624,6 +624,31 @@ public void direct_dependencies_to_self_finds_correct_set_of_origin_types() {
                 .transform(toGuava(GET_ORIGIN_CLASS)).toSet();
     }
 
+    @Test
+    public void stubs_have_empty_dependencies_and_are_not_imported() {
+        class Element {
+        }
+        class DependsOnArray {
+            Element[] array;
+        }
+        JavaClass directlyImportedClass = new ClassFileImporter().importClasses(DependsOnArray.class).get(DependsOnArray.class);
+        JavaClass additionallyImportedClass = directlyImportedClass.getField("array").getRawType();  // Element[]
+        JavaClass stub = additionallyImportedClass.getComponentType();  // Element
+
+        assertThat(stub.getDirectDependenciesFromSelf()).isEmpty();
+        assertThat(stub.getDirectDependenciesToSelf()).isEmpty();
+        assertThat(stub.getFieldsWithTypeOfSelf()).isEmpty();
+        assertThat(stub.getMethodsWithParameterTypeOfSelf()).isEmpty();
+        assertThat(stub.getMethodsWithReturnTypeOfSelf()).isEmpty();
+        assertThat(stub.getMethodThrowsDeclarationsWithTypeOfSelf()).isEmpty();
+        assertThat(stub.getConstructorsWithParameterTypeOfSelf()).isEmpty();
+        assertThat(stub.getConstructorsWithThrowsDeclarationTypeOfSelf()).isEmpty();
+        assertThat(stub.getAnnotationsWithTypeOfSelf()).isEmpty();
+        assertThat(directlyImportedClass.isImported()).isTrue();
+        assertThat(additionallyImportedClass.isImported()).isTrue();
+        assertThat(stub.isImported()).isFalse();
```

Comment:
Maybe make this the first assertion? Would set the tone for me when I read it, i.e. "ah, it is not imported -> ah, now all the dependencies are empty" :wink: